lib/spack/spack/build_environment.py

```diff
@@ -136,7 +136,7 @@ def add_env_path(path):
 
     # Prefixes of all of the package's dependencies go in
     # SPACK_DEPENDENCIES
-    dep_prefixes = [d.prefix for d in pkg.spec.traverse(root=False)]
+    dep_prefixes = [d.prefix for d in pkg.spec.traverse(root=False, deptype=spack.DEP_BUILD)]
```

Comment:
`SPACK_DEPENDENCIES` is actually used in `cc` to set RPATHs so i think you want to change this to `spack.DEP_INSTALL`